app/models/katello/concerns/operatingsystem_extensions.rb

```diff
@@ -11,28 +11,91 @@ module OperatingsystemExtensions
         before_create :set_atomic_attributes, :if => proc { |os| os.name == ::Operatingsystem::REDHAT_ATOMIC_HOST_OS }
       end
 
-      def assign_templates!
-        # Automatically assign default templates
-        if self.family == 'Redhat'
-          TemplateKind.all.each do |kind|
-            if name == ::Operatingsystem::REDHAT_ATOMIC_HOST_OS && kind.name == "provision"
-              provisioning_template_name = Setting["katello_default_atomic_provision"]
-            else
-              provisioning_template_name = Setting["katello_default_#{kind.name}"]
-            end
+      def find_related_os(ignored_os)
+        [true, false].each do |with_major|
+          query = Operatingsystem.where(name: self.name).where.not(id: ignored_os)
+          query = query.where(major: self.major) if with_major
+          related_os = query.order(created_at: :desc).last
+          return related_os unless related_os.nil?
+        end
+        nil
+      end
+
+      def assign_related_os_templates
+        ignored_os = [self.id]
+        3.times do
```

Comment:
This is a good question. My thought was, that its enough to loop throw the configured OS for 3 times if it finds OS but they don't have valid partition/provisioning templates set. Of course, it could loop throw all OS, too.